server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -402,6 +407,11 @@ public boolean shouldCache(Query query) {
         persistMetadata(path, indexSettings, shardRouting, null, logger);
         this.useRetentionLeasesInPeerRecovery = replicationTracker.hasAllPeerRecoveryRetentionLeases();
         this.refreshPendingLocationListener = new RefreshPendingLocationListener();
+        if (indexSettings.isSegRepEnabled() == true) {
+            this.checkpointRefreshListener = new CheckpointRefreshListener(this, checkpointPublisher);
+        } else {
+            this.checkpointRefreshListener = null;
```

Comment:
nit: you don't need the else here, this value will already be null.

Reply:
removed this check as we are passing null